zebra/zebra_nhg.c

```diff
@@ -459,6 +459,9 @@ uint32_t zebra_nhg_id_key(const void *arg)
 static bool nhg_compare_nexthops(const struct nexthop *nh1,
 				 const struct nexthop *nh2)
 {
+	if (!nh1 && !nh2)
+		return false;
+
```

Comment:
How do we ever send in NULL values at all?

Reply:
heh - it's about what coverity thinks might be possible, I suppose? it seemed to see that we null-checked one or the other of nh1, nh2 - but didn't handle the case where both were NULL.

Reply:
I think I asked he question wrong.  Is coverity pointing out that you are only checking for null in some cases but not others, where if you removed the null checks entirely it would be perfectly happy?

In other words is the calling function ever going to pass in a null pointer here?

Reply:
these 'comparison' functions do typically check for the case where one arg is NULL - but we weren't checking for the case where both were NULL. I believe that the callers in this specific case do require that one arg be non-NULL - so I do think we want to handle that case. but coverity ... didn't seem happy with that?

Reply:
actually if you look most of the comparison functions in FRR do not check for one item being null or not.  The assumption is that you would never call it with a null argument.  Hence my question about this.

Reply:
I went through and cleaned up a bunch of functions in the code base doing this type of test that were useless.  I would prefer that we stuck to one way of doing it with the same assumptions for all code.

Reply:
so this gets used in paths where we're comparing two lists of nexthops. it's not guaranteed that the lists are the same length, for example, so at the end of one list we'll have one pointer, and one NULL. are you saying you'd like that specific case to be handled up in the callers of this 'compare' function?

Reply:
returning `false` when both nexthops don't exist is kind of weird. If this change is only to make SA happy, maybe an `assert(nh1 || nh2)` would be better.

Reply:
^ this